core/src/main/java/org/apache/iceberg/BaseRewriteManifests.java

```diff
@@ -248,7 +247,7 @@ private void performRewrite(List<ManifestFile> currentManifests) {
             }
           });
     } finally {
-      Tasks.foreach(writers.values()).executeWith(ThreadPools.getWorkerPool()).run(WriterWrapper::close);
+      Tasks.foreach(writers.values()).executeWith(planExecutorService()).run(WriterWrapper::close);
     }
```

Comment:
I think we should not close it here. It should be closed where it's is provided:
1. if it's provided by calling `planWith`, then it ownership belongs to the caller;
2. if it's the default persistent worker pool like before, then it can not be closed.

Reply:
I agree. The executor service is owned elsewhere.